src/App.tsx

```diff
@@ -1,24 +1,273 @@
+/* eslint-disable jsx-a11y/no-autofocus */
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+/* eslint-disable react-hooks/rules-of-hooks */
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect, useMemo, useRef, useState,
+} from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Status } from './types/Status';
+import { Errors } from './types/Errors';
+import { TodosFilter } from './components/TodoFilter/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 7023;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredType, setFilteredType] = useState(Status.All);
+  const [, setTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [title, setTitle] = useState<string>('');
+
+  const titleRef = useRef<HTMLInputElement | null>(null);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleUpdateTodo = (todo: Todo) => {
+    updateTodo(todo)
+      .then(() => setTodos(current => current.map(t => (t.id === todo.id ? todo : t))))
+      .catch(() => setErrorMessage(Errors.CAN_NOT_UPDATE_TODO));
+  };
+
+  const handleToggleAll = async () => {
```

Comment:
no need to use `async` if you don't use `await` inside
```suggestion
  const handleToggleAll = () => {
```

Reply:
not fixed